src/Aspire.Dashboard/Components/Pages/StructuredLogs.razor

```diff
@@ -16,9 +16,15 @@
 
 <PageTitle><ApplicationName ResourceName="@nameof(Dashboard.Resources.StructuredLogs.StructuredLogsPageTitle)" Loc="@Loc" /></PageTitle>
 
-<div class="logs-layout">
-    <h1 class="page-header">@Loc[nameof(Dashboard.Resources.StructuredLogs.StructuredLogsHeader)]</h1>
-    <FluentToolbar Orientation="Orientation.Horizontal">
+<AspirePageContentLayout
```

Comment:
I noticed that any filter change that changes the page URL causes the filter UI to close.

For example, changing the filter textbox doesn't close the filter UI. Changing the application, level or filters does.

![filter-close](https://github.com/dotnet/aspire/assets/303201/00d264e4-7029-4c60-94c6-fa71d54c8ddc)

Reply:
Also, clicking add filter causes the filter view to close when the sidebar opens.

Reply:
> I noticed that any filter change that changes the page URL causes the filter UI to close. For example, changing the filter textbox doesn't close the filter UI. Changing the application, level or filters does.

Dialogs automatically close without any callback when the URL is changed (@vnbaaij I'm guessing that's not intentional), so I have changed it to apply non-navigation changes only when you close the dialog. But it will be expected to close when switching application

> Also, clicking add filter causes the filter view to close when the sidebar opens.

This was on purpose, since focus is getting messed up when multiple panels are open at the same time.

Reply:
Is the workaround complicated? I'd rather wait for fixes to controls that make the dashboard code more complicated.

Reply:
It's not very complicated, consisting of a block in `IPageWithSessionAndUrlState` that prevents re-render and navigation until after the dialog is closed. But ideally there would be a library solution

Reply:
@vnbaaij additionally, `OnDialogClosing` is not called when calling `IDialogReference.OnCloseAsync`